lib/spack/llnl/util/filesystem.py

```diff
@@ -472,3 +477,152 @@ def find_library_path(libname, *paths):
         if os.path.exists(library):
             return path
     return None
+
+# Utilities for libraries
+
+
+class LibraryList(collections.Sequence):
+    """Sequence of absolute paths to libraries
+
+    Provides a few convenience methods to manipulate library paths and get
+    commonly used compiler flags or names
+    """
+    def __init__(self, libraries):
+        self.libraries = list(libraries)
+
+    @property
+    def directories(self):
+        """Stable de-duplication of the directories where the libraries
+        reside
+
+        >>> l = LibraryList(['/dir1/liba.a', '/dir2/libb.a', '/dir1/libc.a'])
+        >>> assert l.directories == ['/dir1', '/dir2']
+        """
+        return list(dedupe(os.path.dirname(x) for x in self.libraries))
+
+    @property
+    def libnames(self):
```

Comment:
since you changed this function from returning `libFOO` to `libFOO.a`, maybe call it `basenames` which is self-explanatory and avoids confusion that `FOO` (names of the libraries) is returned.